pkg/oauth/external/handler.go

```diff
@@ -139,6 +139,18 @@ func (h *Handler) AuthenticatePassword(ctx context.Context, username, password s
 	return identitymapper.ResponseFor(h.mapper, identity)
 }
 
+// sanitizeURL removes possible sensitive data from the URL, useful for logging
+func sanitizeURL(u *url.URL) *url.URL {
```

Comment:
Functions are copying the value, so having a `url.URL` struct as an argument instead of a pointer, would take that part off you. Except of the `URL.User`,  as you do below already.

Reply:
fixed